hermes-management/src/main/java/pl/allegro/tech/hermes/management/infrastructure/kafka/service/KafkaConsumerGroupManager.java

```diff
@@ -0,0 +1,111 @@
+package pl.allegro.tech.hermes.management.infrastructure.kafka.service;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import kafka.common.TopicAndPartition;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.common.TopicPartition;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import pl.allegro.tech.hermes.api.Subscription;
+import pl.allegro.tech.hermes.common.broker.BrokerDetails;
+import pl.allegro.tech.hermes.common.broker.BrokerStorage;
+import pl.allegro.tech.hermes.common.kafka.ConsumerGroupId;
+import pl.allegro.tech.hermes.common.kafka.KafkaNamesMapper;
+import pl.allegro.tech.hermes.management.domain.subscription.ConsumerGroupManager;
+
+import java.util.Properties;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.stream.IntStream;
+
+import static org.apache.kafka.clients.consumer.ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG;
+import static org.apache.kafka.clients.consumer.ConsumerConfig.GROUP_ID_CONFIG;
+
+public class KafkaConsumerGroupManager implements ConsumerGroupManager {
+
+    private final Logger logger = LoggerFactory.getLogger(KafkaConsumerGroupManager.class);
+
+    private final BrokerStorage brokerStorage;
+    private final KafkaNamesMapper kafkaNamesMapper;
+    private final Integer partitions;
+    private final String clusterName;
+    private final ExecutorService executor;
+
+    public KafkaConsumerGroupManager(BrokerStorage brokerStorage,
+                                     KafkaNamesMapper kafkaNamesMapper,
+                                     Integer partitions,
+                                     String clusterName) {
+        this.brokerStorage = brokerStorage;
+        this.kafkaNamesMapper = kafkaNamesMapper;
+        this.partitions = partitions;
```

Comment:
Actually, this is the current number of partitions we want newly created topics to have.
Old topics that were created some time ago, or are some manually created, may have different number of partitions. It would be best to get the number from the exact kafka topic.

Reply:
Thanks, I fixed it.